A3A/addons/core/functions/FactoryOverhaul/fn_productionGUIControls.sqf

```diff
@@ -0,0 +1,192 @@
+		lbAdd [IDC_FACTORY_WEPS, _itemName];
+	} forEach call FO_fn_getUnlockedItems;
+};
+
+
+FO_fn_GUI_lookupFactoryIDbyName = {
+    params ["_factoryTownPairs","_text"];
+
+    private _selectedFactoryFromTownname = "";
+    {
+        if ((_x select 1) == _text) then {
+            _selectedFactoryFromTownname = _x select 0;
+        };
+    } forEach _factoryTownPairs;
+//systemChat str _selectedFactoryFromTownname;
+
+_selectedFactoryFromTownname;
+};
+
+/////////////////////////////////////////// TODO
+FO_fn_GUI_lookupItemClassbyName = {
+    params ["_unlockedItemsArray","_displayName"];
+
+    private _selectedItemClassname = "";
+    {
+        if ((_x select 1) == _text) then {
+            _selectedItemClassname = _x select 0;
+        };
+    } forEach _factoryTownPairs;
+//systemChat str _selectedFactoryFromTownname;
+
+_selectedItemClassname;
+};
+/////////////////////////////////////////// TODO
+
+
+FN_FO_GUI_addEventHandlers = {
+    private _display = findDisplay IDX_FACTORY_GUI;
+    private _listBoxFactories = _display displayCtrl IDC_FACTORY_LIST;
+    private _StartStopButton = _display displayCtrl IDC_STARTSTOP_BTN;
+    private _exitButton = _display displayCtrl IDC_EXIT_BUTTON;
+    
+    //systemChat "Selection changed";
+
+
+    // Add the Selection Changed event handler to Owned Factories ListBox
+    _listBoxFactories ctrlAddEventHandler ["LBSelChanged", {
+        private _selIndex = lbCurSel IDC_FACTORY_LIST;
+        private _selText = lbText [IDC_FACTORY_LIST, _selIndex];
+      
+        // Select the town name in the GUI -> Convert it to factory name -> Get that factory's structure.
+        private _ownedFactories = call FO_fn_GUI_createFactoryTownPairs;
+
+        private _factoryID = [_ownedFactories, _selText] call FO_fn_GUI_lookupFactoryIDbyName;
+        
+        systemChat format ["%1 factory selected, ID %2", _selText, _factoryID];
+
+        // [factory, item, amount, progress, progressMax, magazine]
+        // Get the data to display to the user.
+        private _structure = _factoryID call FO_fn_getFactoryStructure;
+        //systemChat str _structure;
+
+        if (([_structure] call FO_fn_factoryIsProducing) isEqualTo true) then {
+            private _item = _structure select 1;
+            private _itemDisplayName = getText (configFile >> "CfgWeapons" >> _item >> "displayName");
+            private _qty = str (_structure select 2);
+            ctrlSetText [IDC_ITEM, _itemDisplayName];
+            ctrlSetText [IDC_QTY, _qty];
+
+            ctrlSetText [IDC_STARTSTOP_BTN, "Stop Production"];
+
+        } else {
+            ctrlSetText [IDC_ITEM, ""];
+            ctrlSetText [IDC_QTY, ""];
+            ctrlSetText [IDC_STARTSTOP_BTN, "Start Production"];
+        };
+
+    }];
+
+        // Add the ButtonClick event handler to Start/Stop Production Button.
+    _StartStopButton ctrlAddEventHandler ["ButtonClick", {
```

Comment:
A lot of this is basically the same as what's written higher up here, could reduce the duplication with more functions, but leave that for later.